internal/pkg/cli/app_init.go

```diff
@@ -299,15 +314,41 @@ func (o *initAppOpts) askDockerfile() error {
 }
 
 func (o *initAppOpts) askAppPort() error {
+	// Use flag before anything else
 	if o.AppPort != 0 {
 		return nil
 	}
 
+	log.Infof(fmtParsePortFromDockerfileStart, o.DockerfilePath, o.AppName)
+
+	o.setupParser(o)
+	ports := o.df.GetExposedPorts()
+
+	var defaultPort = defaultAppPortString
+	switch len(ports) {
+	case 0:
+		log.Infof(fmtParsePortFromDockerfileFailedNoPort, o.AppName)
+	case 1:
+		// catch the case where we couldn't validate a port from dockerfile (0 or invalid)
+		if ports[0] == 0 {
```

Comment:
I'm a little confused by this case - is this when there's like, tcp or udp or other weirdness in the port?

Reply:
This is because 0 isn't really a port--it stands for "connect on any available port" which is a legacy behavior that nobody really uses anymore since it's nearly undocumentable, and we use it here as "couldn't parse" because our ports are represented as `uint16`s. I think you're right that once I add some error handling it'll clarify things, at the very least by returning an error and an empty slice

Reply:
this is when the port in the dockerfile is out of bounds or couldn't be correctly parsed into an int. Don't know that that second case will ever happen though.